### PR TITLE
fix(Item_Operatingsystem): update / handle one OS like GLPI

### PR DESCRIPTION
**Pull request by stonebuzz**
When importing computers with DataInjection, the operating system is added, not replaced. This results in 2 or more OS per computer, which should even not be possible in GLPI.

This PR fix this

fix : internal ref : 30448

### CODE REVIEW

File: inc/item_operatingsysteminjection.class.php [hunk 1]
```diff
@@ -92,6 +92,25 @@ function addOrUpdateObject($values = [], $options = []) {
       return $lib->getInjectionResults();
    }
 
+
+   /**
+    * @param $injectionClass
+    * @param $values
+    * @param $options
```
Comment (trasher) [inline inc/item_operatingsysteminjection.class.php:99]:
```suggestion
    * @param ?? $injectionClass
    * @param array $values
    * @param array $options
```

Comment (stonebuzz) [reply inc/item_operatingsysteminjection.class.php]: fixed 
```$injectionClass``` is ```PluginDatainjectionInjectionInterface```

File: inc/item_operatingsysteminjection.class.php [hunk 2]
```diff
@@ -92,6 +92,25 @@ function addOrUpdateObject($values = [], $options = []) {
...
+    * @param PluginDatainjectionInjectionInterface $injectionClass
+    * @param array $values
+    * @param array $options
+   **/
+  function customDataAlreadyInDB($injectionClass, $values, $options) {
+      $item_operatingsystem = new Item_OperatingSystem();
+      if (!$item_operatingsystem->getFromDBByCrit([
+         'itemtype' => $values['itemtype'],
+         'items_id' => $values['items_id'],
+      ])) {
```
Comment (cedric-anne) [inline inc/item_operatingsysteminjection.class.php:106]: `getFromDBByCrit` will return false if there are already mutiple OS entries, so the problem will not be solved for people that already used the plugin to inject multiple OS on a computer.

**Review by cedric-anne — CHANGES_REQUESTED** [review]

File: inc/item_operatingsysteminjection.class.php (hunk 2 above)
Comment (stonebuzz) [reply inc/item_operatingsysteminjection.class.php]: last commit handle this case (like ```Domain``` from native inventory)

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**